gallery/src/PageHeader/HostedVersionsSelect.tsx

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright (c) 2019-present Sonatype, Inc.
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
+ */
+import React, { useEffect, useState } from 'react';
+import { without } from 'ramda';
+import { NxStatefulDropdown } from '@sonatype/react-shared-components';
+import compareVersions from 'compare-versions';
+
+import packageJson from '../../package.json';
+
+import './HostedVersionsSelect.scss';
+
+const CURRENT_VERSION = packageJson.version;
+const HOSTED_VERSIONS_JSON_PATH = '/hosted-versions.json';
+
+// eslint-disable-next-line no-undef
+const PRODUCTION = process.env.PRODUCTION;
+
+const getHostedVersions = async () => {
+  const response = await fetch(HOSTED_VERSIONS_JSON_PATH);
+
+  if (!response.ok) {
+    throw new Error('Failed to load hosted versions.');
+  }
+
+  return response.json();
+};
+
+const HostedVersionsSelect = () => {
+  const [isLoadingHostedVersions, setIsLoadingHostedVersions] = useState(false);
+  const [hasError, setHasError] = useState(false);
+  const [hostedVersions, setHostedVersions] = useState([CURRENT_VERSION]);
+
+  useEffect(() => {
+    setIsLoadingHostedVersions(true);
+
+    const loadHostedVersions = async () => {
+      try {
+        const hostedVersions = await getHostedVersions();
+        const sortedVersions = without(['latest'], hostedVersions).sort(compareVersions).reverse();
+
+        setHostedVersions(sortedVersions);
+        setIsLoadingHostedVersions(false);
+      }
+      catch {
+        setHasError(true);
+        setIsLoadingHostedVersions(false);
+      }
+    };
+
+    loadHostedVersions();
+  }, []);
+
+  return (
+    <div className="gallery-hosted-versions-select">
+      <span className="gallery-hosted-versions-select__version">
+        Version { (!PRODUCTION || hasError) && CURRENT_VERSION }
+      </span>
+      {
+        PRODUCTION && !isLoadingHostedVersions && !hasError && (
+          <NxStatefulDropdown className="gallery-hosted-versions-select__dropdown nx-dropdown--short"
```

Comment:
Where can I see/test this dropdown now that you've got it so that it doesn't appear locally?

Reply:
> Where can I see/test this dropdown now that you've got it so that it doesn't appear locally?

Currently, the only way to test it locally is to manually change
`HOSTED_VERSIONS_JSON_PATH` from  `/hosted-versions.json` to `https://gallery.sonatype.dev/hosted-versions.json` and change `IS_PRODUCTION` to `true`.

Should I create a flag for this to test locally? I was also planning to do a build and I think it should upload to that /versions so we can test it live

Reply:
> I was also planning to do a build and I think it should upload to that /versions so we can test it live

Oh that should work

Reply:
> > I was also planning to do a build and I think it should upload to that /versions so we can test it live
> 
> Oh that should work

Looks to be working here! https://gallery.sonatype.dev/versions/10.0.3/#/